src/systempolicy/systemPolicy.go

```diff
@@ -286,29 +287,83 @@ func WriteSystemPoliciesToFile(namespace, clustername, labels string) {
 	WriteSystemPoliciesToFile_Ext(namespace, clustername, labels)
 }
 
-func GetSysPolicy(namespace, clustername, labels string) *wpb.WorkerResponse {
+func GetSysPolicy(namespace, clustername, labels, fromsource string) *wpb.WorkerResponse {
 
-	sysPols := populateKnoxSysPolicyFromWPFSDb(namespace, clustername, labels)
-	kubearmorPolicies := plugin.ConvertKnoxSystemPolicyToKubeArmorPolicy(sysPols)
+	kubearmorK8SPolicies := extractK8SSystemPolicies(namespace, clustername, labels)
+	kubearmorVMPolicies := extractVMSystemPolicies(types.PolicyDiscoveryVMNamespace, clustername, labels, fromsource)
 
 	var response wpb.WorkerResponse
-	for i := range kubearmorPolicies {
+
+	// system policy for k8s
+	for i := range kubearmorK8SPolicies {
 		kubearmorpolicy := wpb.KubeArmorPolicy{}
 
-		val, err := json.Marshal(&kubearmorPolicies[i])
+		val, err := json.Marshal(&kubearmorK8SPolicies[i])
 		if err != nil {
 			log.Error().Msg(err.Error())
 		}
 		kubearmorpolicy.Data = val
 
 		response.Kubearmorpolicy = append(response.Kubearmorpolicy, &kubearmorpolicy)
 	}
+
+	// system policy for VM
+	for i := range kubearmorVMPolicies {
+		kubearmorpolicy := wpb.KubeArmorPolicy{}
+
+		val, err := json.Marshal(&kubearmorVMPolicies[i])
+		if err != nil {
+			log.Error().Msg(err.Error())
```

Comment:
It is better to `continue` in this failure case. Also please add a text label before the error message. 
```suggestion
			log.Error().Msg("kubearmorVMPolicy json marshal failed err=%v", err.Error())
			continue
```